src/librustc_codegen_llvm/type_of.rs

```diff
@@ -4,7 +4,7 @@ use crate::type_::Type;
 use rustc::ty::{self, Ty, TypeFoldable};
 use rustc::ty::layout::{self, Align, LayoutOf, FnTypeExt, PointeeInfo, Size, TyLayout};
 use rustc_target::abi::{FloatTy, TyLayoutMethods};
-use rustc_mir::monomorphize::item::DefPathBasedNames;
+use rustc_monomorphize::item::DefPathBasedNames;
```

Comment:
Okay, I looked at this, and we can probably move `DefPathBasedNames` fully to `ty::print`, in a submodule named `obsolete`, and open an issue about refactoring it to use `ty::print` infra (I already left comments, I believe, but they do not have an issue attached).

Reply:
Oh and once we do that, `MonoItemExt` can just be inherent methods on `MonoItem`.
Also, I'm a bit surprised `MonoItem::GlobalAsm` uses a `HirId` and not a `DefId`, that's weird.

Reply:
@eddyb It looks like `librustc_codegen_ssa` also uses `MonoItemExt` :(

https://github.com/rust-lang/rust/blob/27cc0db7a248308fc2634ac68d7608a20b4a1c09/src/librustc_codegen_ssa/mono_item.rs#L10-L12

Moving those methods to `rustc::mir::mono` is hard because they use `trait rustc_codegen_ssa::traits::BuilderMethods`, which itself uses half a dozen other traits from the that crate.

We could try to move the transitive closure of all of those traits to `librustc`, but things could get messy... I don't know if it will work.

I have yet to investigate `CodegenUnitExt`...

Reply:
I am talking about `rustc_mir::monomorphize::item::MonoItemExt`, not `rustc_codegen_ssa::mono_item::MonoItemExt` - you can ignore the latter, it's part of codegen extensions to non-codegen types.

Reply:
@eddyb `rustc_codegen_ssa::mono_item::MonoItemExt: rustc_mir::monomorphize::item::MonoItemExt`

Reply:
That's only for the trick of being able to put the body of the methods in the trait, as default methods, by relying on `as_mono_item`, the methods will still obviously be available on `MonoItem`, they're not going away, so the trait isn't needed.

Reply:
Heh, sorry my brain was missing in action.